src/org/thoughtcrime/securesms/InstantOnboardingActivity.java

```diff
@@ -199,6 +199,24 @@ public void onStart() {
     }
   }
 
+  @Override
+  protected void onPause() {
+    super.onPause();
+
+    final String displayName = name.getText().toString();
+    DcHelper.set(this, DcHelper.CONFIG_DISPLAY_NAME, TextUtils.isEmpty(displayName)? null : displayName);
+
+    if (avatarChanged) {
+      try {
+        AvatarHelper.setSelfAvatar(InstantOnboardingActivity.this, avatarBmp);
+        Prefs.setProfileAvatarId(InstantOnboardingActivity.this, new SecureRandom().nextInt());
+        avatarChanged = false;
+      } catch (IOException e) {
+        Log.e(TAG, "Failed to save avatar", e);
+      }
+    }
```

Comment:
one could factor out these lines as they're similar to createProfile(), however, due to different threads and additional ongoings in createProfile()  it is probably not worth the effort